src/options.ts

```diff
@@ -1,5 +1,5 @@
 import {ScenegraphData} from './options';
-export interface Option {
+export interface Option extends Partial<Record<string, string | object | boolean | number>> {
```

Comment:
Can you explain why this is necessary?

Reply:
It's necessary to make the string index signature optional because otherwise all the concrete fields are flagged as not being compatible with the string index signature:

```
node_modules/vega-tooltip/build/src/options.d.ts(3,5): error TS2411: Property 'showAllFields' of type 'boolean | undefined' is not assignable to string index type 'string | number | boolean | object'.

3     showAllFields?: boolean;
      ~~~~~~~~~~~~~~~~~~~~~~~~


node_modules/vega-tooltip/build/src/options.d.ts(4,5): error TS2411: Property 'fields' of type 'FieldOption[] | undefined' is not assignable to string index type 'string | number | boolean | object'.

4     fields?: FieldOption[];
      ~~~~~~~~~~~~~~~~~~~~~~~


node_modules/vega-tooltip/build/src/options.d.ts(5,5): error TS2411: Property 'delay' of type 'number | undefined' is not assignable to string index type 'string | number | boolean | object'.

5     delay?: number;
      ~~~~~~~~~~~~~~~


node_modules/vega-tooltip/build/src/options.d.ts(6,5): error TS2411: Property 'offset' of type '{ x: number; y: number; } | undefined' is not assignable to string index type 'string | number | boolean | object'.

6     offset?: {
      ~~~~~~~~~~7         x: number;
  ~~~~~~~~~~~~~~~~~~8         y: number;
  ~~~~~~~~~~~~~~~~~~9     };
  ~~~~~~


node_modules/vega-tooltip/build/src/options.d.ts(10,5): error TS2411: Property 'onAppear' of type '((event: Event, item: object) => void) | undefined' is not assignable to string index type 'string | number | boolean | object'.

10     onAppear?(event: Event, item: object): void;
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


node_modules/vega-tooltip/build/src/options.d.ts(11,5): error TS2411: Property 'onMove' of type '((event: Event, item: object) => void) | undefined' is not assignable to string index type 'string | number | boolean | object'.

11     onMove?(event: Event, item: object): void;
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


node_modules/vega-tooltip/build/src/options.d.ts(12,5): error TS2411: Property 'onDisappear' of type '((event: Event, item: object) => void) | undefined' is not assignable to string index type 'string | number | boolean | object'.

12     onDisappear?(event: Event, item: object): void;
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


node_modules/vega-tooltip/build/src/options.d.ts(13,5): error TS2411: Property 'colorTheme' of type '"light" | "dark" | undefined' is not assignable to string index type 'string | number | boolean | object'.

13     colorTheme?: 'light' | 'dark';
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
I think the code is bad here. We should remove the `[key: string]` stuff or fix that.

Reply:
```ts
export type Option = {
  showAllFields?: boolean;
  fields?: FieldOption[];
  delay?: number;
  offset?: {
    x: number,
    y: number
  };
  onAppear?(event: Event, item: object): void;
  onMove?(event: Event, item: object): void;
  onDisappear?(event: Event, item: object): void;
  colorTheme?: 'light' | 'dark';
} & {
  [key: string]: string | object | boolean | number
};
```

would also be cleaner

Reply:
I will check with @sirahd and fix it.

Reply:
Even though we remove this, there is still some other part of the code that doesn't pass the `strictNullCheck` flag. If you want it to compile under that flag, it'll need a little bit of work.

Reply:
Let's fix this one as it indicates a problem. The other issues can come later.